contracts/Distributor.sol

```diff
@@ -0,0 +1,148 @@
+pragma solidity 0.5.7;
+
+import * as ERC721 from "@openzeppelin/contracts/token/ERC721/ERC721Full.sol";
+import * as IERC20 from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import * as Ownable from "@openzeppelin/contracts/ownership/Ownable.sol";
+import * as SafeMath from "./external/openzeppelin-solidity/math/SafeMath.sol";
+import * as INXMMaster from "./INXMMaster.sol";
+import * as Pool1 from "./Pool1.sol";
+import * as PoolData from "./PoolData.sol";
+import * as Claims from "./Claims.sol";
+import * as NXMToken from "./NXMToken.sol";
+import * as QuotationData from "./QuotationData.sol";
+
+contract Distributor is ERC721.ERC721Full("NXMDistributorNFT", "NXMDNFT"), Ownable.Ownable {
+
+  struct TokenData {
+    uint expirationTimestamp;
+    address lastOwner;
+    bytes4 coverCurrency;
+    uint[] coverDetails;
+    uint coverId;
+  }
+
+  uint public constant CLAIM_VALIDITY_MAX_DAYS_OVER_COVER_PERIOD = 30 days;
```

Comment:
it should be read from here: https://github.com/danoctavian/smart-contracts/blob/8d2eaf6d322ace4bdf31879114a4c55f09041fe9/contracts/TokenData.sol#L26